core/corehttp/ipns_hostname.go

```diff
@@ -19,8 +19,8 @@ func IPNSHostnameOption() ServeOption {
 			defer cancel()
 
 			host := strings.SplitN(r.Host, ":", 2)[0]
-			if p, err := n.Namesys.Resolve(ctx, host); err == nil {
-				r.URL.Path = "/ipfs/" + p.String() + r.URL.Path
+			if p, err := n.Namesys.Resolve(ctx, host, 0); err == nil {
+				r.URL.Path = p.String() + r.URL.Path
```

Comment:
I'd prefer the second approach because we don't duplicate the logic here. We just rewrite to a IPNS path which is handled by other logic which also pulls the DNS record and resolves in a better way than we're doing with the first approach. Also for `/ipns/` there are special handlings defined in delivery which would need to be extended for this HostnameOption thing.